From: =?UTF-8?q?Mika=20Pfl=C3=BCger?= <debian@mikapflueger.de>
Date: Fri, 2 Mar 2012 02:36:13 +0100
Subject: tmpreaper policy: Some debian customisations, also make tmpreaper
 policy MCS-aware (add interface in mcs.if for that)

---
 policy/modules/admin/tmpreaper.fc |    4 ++++
 policy/modules/admin/tmpreaper.te |    4 ++--
 policy/modules/kernel/mcs.if      |   20 ++++++++++++++++++++
 3 files changed, 26 insertions(+), 2 deletions(-)

diff --git a/policy/modules/admin/tmpreaper.fc b/policy/modules/admin/tmpreaper.fc
index 81077db..0dfb148 100644
--- a/policy/modules/admin/tmpreaper.fc
+++ b/policy/modules/admin/tmpreaper.fc
@@ -1,2 +1,6 @@
 /usr/sbin/tmpreaper		--	gen_context(system_u:object_r:tmpreaper_exec_t,s0)
 /usr/sbin/tmpwatch		--	gen_context(system_u:object_r:tmpreaper_exec_t,s0)
+ifdef(`distro_debian', `
+/etc/init\.d/mountall-bootclean.sh --	gen_context(system_u:object_r:tmpreaper_exec_t,s0)
+/etc/init\.d/mountnfs-bootclean.sh --	gen_context(system_u:object_r:tmpreaper_exec_t,s0)
+')
diff --git a/policy/modules/admin/tmpreaper.te b/policy/modules/admin/tmpreaper.te
index 6a5004b..ac9977d 100644
--- a/policy/modules/admin/tmpreaper.te
+++ b/policy/modules/admin/tmpreaper.te
@@ -30,8 +30,7 @@ files_setattr_all_tmp_dirs(tmpreaper_t)
 files_getattr_all_dirs(tmpreaper_t)
 files_getattr_all_files(tmpreaper_t)
 
-mls_file_read_all_levels(tmpreaper_t)
-mls_file_write_all_levels(tmpreaper_t)
+mcs_file_delete_all(tmpreaper_t)
 
 logging_send_syslog_msg(tmpreaper_t)
 
@@ -39,6 +38,7 @@ miscfiles_read_localization(tmpreaper_t)
 miscfiles_delete_man_pages(tmpreaper_t)
 
 cron_system_entry(tmpreaper_t, tmpreaper_exec_t)
+init_system_domain(tmpreaper_t, tmpreaper_exec_t)
 
 ifdef(`distro_redhat',`
 	userdom_list_user_home_content(tmpreaper_t)
diff --git a/policy/modules/kernel/mcs.if b/policy/modules/kernel/mcs.if
index f52faaf..66cbb76 100644
--- a/policy/modules/kernel/mcs.if
+++ b/policy/modules/kernel/mcs.if
@@ -45,6 +45,26 @@ interface(`mcs_file_write_all',`
 
 ########################################
 ## <summary>
+##	This domain is allowed to delete files and directories
+##	regardless of their MCS category set.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain target for user exemption.
+##	</summary>
+## </param>
+## <rolecap/>
+#
+interface(`mcs_file_delete_all',`
+	gen_require(`
+		attribute mcsdeleteall;
+	')
+
+	typeattribute $1 mcsdeleteall;
+')
+
+########################################
+## <summary>
 ##	This domain is allowed to sigkill and sigstop 
 ##	all domains regardless of their MCS category set.
 ## </summary>
